crates/proto/proto/requests.proto

```diff
@@ -9,9 +9,8 @@ import "note.proto";
 
 message AccountUpdate {
     account.AccountId account_id = 1;
-    digest.Digest account_hash = 2;
-    // Details for public (on-chain) account.
-    optional bytes details = 3;
+    digest.Digest final_state_hash = 2;
+    bytes details = 3;
 }
 
 message ApplyBlockRequest {
```

Comment:
I'm wondering if we should change `ApplyBlockRequest` message to be just:
```proto
message ApplyBlockRequest {
    bytes block = 1
}
```
And the payload would be just the serialized version of the `Block` struct.

The main motivation is that it should reduce the complexity of handling different domain objects (e.g., we just reuse domain objects from `miden-base` instead of having protobuf messages and local domain objects for them).